### PR TITLE
Add phone confirmation page for register journey

### PR DESCRIPTION
**Pull request by wilpow-github**
### Context

As an overseas qualified teacher
I want to sign in/register a Teaching services account
So that I can apply for QTS

### Changes proposed in this pull request

Add a new page at /sign-in/register/phone-confirmation matching the designs in the prototype

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [x] Cleaned commit history
-   [x] Tested by running locally
-   [x] Reminder created to manually clean any removed app settings post deployment

### CODE REVIEW

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Register/EmailConfirmation.cshtml.cs [hunk 1]
```diff
@@ -46,4 +47,14 @@ public async Task<IActionResult> OnPost()
 
         return Redirect(_linkGenerator.RegisterPhone());
     }
+
+    public override void OnPageHandlerExecuting(PageHandlerExecutingContext context)
+    {
+        var authenticationState = context.HttpContext.GetAuthenticationState();
+
+        if (!authenticationState.EmailAddressSet)
```
Comment (wilpow-github) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Register/EmailConfirmation.cshtml.cs:55]: Added this to email confimation as it seemed sensible even though we are not quite sure how to measure milestones in this journey

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Register/PhoneConfirmation.cshtml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            <h1 class="govuk-heading-xl">@ViewBag.Title</h1>
+
+            <p>We’ve sent you a text message with a security code to <b data-testid="mobileNumber">@Model.MobileNumber</b></p>
+
+            <govuk-input asp-for="Code" input-class="govuk-!-width-one-quarter" pattern="[0-9]*" inputmode="numeric" autocomplete="one-time-code">
+                <govuk-input-label class="govuk-label--s" />
+            </govuk-input>
+
+            <p>
+                <a href="@LinkGenerator.RegisterResendPhoneConfirmation()">I have not received a text message</a>
```
Comment (wilpow-github) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Register/PhoneConfirmation.cshtml:23]: This doesn't exist in the register email confirmation page

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Register/PhoneConfirmation.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        IUserVerificationService userConfirmationService,
+        PinValidator pinValidator,
+        IIdentityLinkGenerator linkGenerator)
+        : base(userConfirmationService, pinValidator)
+    {
+        _linkGenerator = linkGenerator;
+    }
+
+    [BindProperty]
+    [Display(Name = "Security code")]
```
Comment (wilpow-github) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Register/PhoneConfirmation.cshtml.cs:22]: There's no consistent name for these codes

File: dotnet-authserver/tests/TeacherIdentity.AuthServer.Tests/AuthenticationStateHelper.cs [hunk 4]
```diff
@@ -116,6 +116,13 @@ internal Configure(HostFixture hostFixture)
                 s.OnEmailVerified(user);
             };
 
+        public Func<AuthenticationState, Task> MobileNumberSet(string? mobileNumber = null) =>
+            s =>
+            {
+                s.OnMobileNumberSet(mobileNumber ?? Faker.Phone.Number());
+                return Task.CompletedTask;
+            };
+
```
Comment (gunndabad) [inline dotnet-authserver/tests/TeacherIdentity.AuthServer.Tests/AuthenticationStateHelper.cs:125]:
```suggestion
        public Func<AuthenticationState, Task> MobileNumberSet(string? email, string? mobileNumber = null) =>
            async s =>
            {
                await EmailVerified(email)(s);
                s.OnMobileNumberSet(mobileNumber ?? Faker.Phone.Number());
            };

```

**Review by gunndabad — APPROVED** [review]

**Pull request merged**